pkg/network/protocols/postgres/telemetry.go

```diff
@@ -22,18 +22,96 @@ const (
 	numberOfBucketsSmallerThanMaxBufferSize = 3
 )
 
+type counterStateEnum int
+
+const (
+	tableAndOperation counterStateEnum = iota + 1
+	operationNotFound
+	tableNameNotFound
+	tableAndOpNotFound
+)
+
+func (e counterStateEnum) String() string {
+	switch e {
+	case tableAndOperation:
+		return "tableAndOperation"
+	case operationNotFound:
+		return "operationNotFound"
+	case tableNameNotFound:
+		return "tableNameNotFound"
+	case tableAndOpNotFound:
+		return "tableAndOpNotFound"
+	default:
+		return fmt.Sprintf("%d", int(e))
+	}
+}
+
+// extractionFailureCounter stores counter when goal was achieved and counter when target not found.
+type extractionFailureCounter struct {
+	// countTableAndOperationFound counts the number of successfully retrieved table name and operation.
+	countTableAndOperationFound *libtelemetry.Counter
+	// countOperationNotFound counts the number of unsuccessful fetches of the operation.
+	countOperationNotFound *libtelemetry.Counter
+	// countTableNameNotFound counts the number of unsuccessful fetches of the table name.
+	countTableNameNotFound *libtelemetry.Counter
+	// countTableAndOpNotFound counts the number of failed attempts to fetch both the table name and the operation.
+	countTableAndOpNotFound *libtelemetry.Counter
+}
+
+// newExtractionFailureCounter creates and returns a new instance
+func newExtractionFailureCounter(metricGroup *libtelemetry.MetricGroup, metricName string, tags ...string) *extractionFailureCounter {
+	return &extractionFailureCounter{
+		countTableAndOperationFound: metricGroup.NewCounter(metricName, append(tags, "state:table_and_op")...),
+		countOperationNotFound:      metricGroup.NewCounter(metricName, append(tags, "state:no_operation")...),
+		countTableNameNotFound:      metricGroup.NewCounter(metricName, append(tags, "state:no_table_name")...),
+		countTableAndOpNotFound:     metricGroup.NewCounter(metricName, append(tags, "state:no_table_no_op")...),
+	}
+}
+
+// inc increments the counter of completed results or counter incomplete results.
+func (c *extractionFailureCounter) inc(state counterStateEnum) {
+	switch state {
+	case tableAndOperation:
+		c.countTableAndOperationFound.Add(1)
+	case operationNotFound:
+		c.countOperationNotFound.Add(1)
+	case tableNameNotFound:
+		c.countTableNameNotFound.Add(1)
+	case tableAndOpNotFound:
+		c.countTableAndOpNotFound.Add(1)
+	default:
+		log.Errorf("postgres query length counter increment function received undefined state: %v", state)
```

Comment:
```suggestion
		log.Errorf("unable to increment extractionFailureCounter due to undefined state: %v", state)
```

Reply:
changed as suggested

Reply:
You have committed it wrong and now you have format issues